diagnostic.js

```diff
@@ -27,33 +27,124 @@
 const mongoose = require('mongoose');
 mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud-diagnostic');
+const db = mongoose.connection;
 
 const Movement = require('./models/movement.js');
 
-/// ADD YOUR CODE BELOW
+const done = function() {
+  db.close();
+};
+
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    'name': name,
+    'description': description,
+    'startYear': startYear,
+    'endYear': endYear
+  })
+  .then(function(movement){
+    console.log(movement);
+  })
+  .catch(console.error())
+  .then(done);
+};
+
+
+const index = () => {
+  Movement.find()
+   .then(function(movements){
+     movements.forEach(function(movement) {
+       console.log(movement.toJSON());
+     });
+   }
+ )
+ .catch(console.error)
+ .then(done);
+};
+
+
+const show = (id) => {
+  Movement.findById(id)
+    .then(function(movement){
+      console.log(movement.toJSON());
+    }
+  )
+  .catch(console.error())
+  .then(done);
+};
+
+
+const update = (id, field, value) => {
+  let modify = {};
+  modify[field] = value;
+  Movement.findById(id)
+    .then(function(movement){
+      movement[field] = value;
+      return movement.save();
+    }
+  )
+  .then(function(movement){
+    console.log(movement.toJSON());
+  })
+  .catch(console.error())
+  .then(done);
+};
+
+
+const destroy = (id) => {
+  Movement.findById(id)
+    .then(function(movement){
+      return movement.remove();
```

Comment:
Missing console.log('removed'); after successfully destroying a document and missing console.log('Not found'); when no document has been found.